### PR TITLE
fix: use 'descendant' descriptor instead of 'child'

### PR DESCRIPTION
**Pull request by danieljhegeman**
## Reason for Change

- #742

## Changes

In 1) code, and 2) error messages:
- Replace `child` with `descendant` where appropriate
- Replace `children` with `descendants` where appropriate

I do not see any cases of function name changes that would be used via import in other repos (the one function whose name has been changed is prefaced with an underscore)

## Testing

- Unit tests updated accordingly
```
pip install git+https://github.com/chanzuckerberg/single-cell-curation/@main#subdirectory=cellxgene_schema_cli
```

## Notes for Reviewer

### CODE REVIEW

File: schema/5.1.0/schema.md
Comment (danieljhegeman) [inline schema/5.1.0/schema.md:1]: @brianraymor feel free to review since this schema file is being edited. Are there potential related changes / conflicts that might occur in other places (beyond this repo, perhaps in non-code public docs?) as a result of this change?

Comment (brianraymor) [reply schema/5.1.0/schema.md:1]: Updating the changelog text for schema version 5.0.0 is questionable, since the editorial change was not introduced until 5.1.0.

> Are there potential related changes / conflicts that might occur in other places (beyond this repo, perhaps in non-code public docs?) as a result of this change?

I don't see any references to "child" or "children" in related documentation such as https://cellxgene.cziscience.com/docs/032__Contribute%20and%20Publish%20Data.

CC: @jahilton

**Review by nayib-jose-gloria — APPROVED** [review]
I'd wait on Brian's approval for the schema.md changes, but the rest of the refactor looks good to me

**Review by brianraymor — APPROVED** [review]

**Pull request merged**
